src/main/java/edu/cornell/kfs/sys/service/impl/AwsSecretServiceImpl.java

```diff
@@ -33,9 +36,66 @@
     protected String kfsSharedNamespace;
     protected int retryCount;
     
+    private Map<String, String> awsSecretsCache;
+    
+    @Override
+    public void clearCache() {
+        awsSecretsCache = new HashMap<String, String>();
```

Comment:
Depending on how long each service instance or its cache is expected to live for, and on whether a given service instance could potentially be used by multiple threads, it might be better to use a ConcurrentHashMap instead of a regular HashMap. (Note, though, that ConcurrentHashMap does not allow null keys or null values.)

Also, this line could be simplified to use "<>()" instead of "<String, String>()"

Reply:
OK, I switched to use ConcurrentHashMap and added logic checking for null values before adding to the cache